src/test/java/htsjdk/beta/codecs/variants/vcf/HtsVCFCodecTest.java

```diff
@@ -105,15 +105,6 @@ public void testRoundTripVCFThroughStream(final IOPath inputPath, final HtsVersi
         }
     }
 
-    @Test(expectedExceptions = IllegalArgumentException.class)
-    public void testRejectWritingV43HeaderAsV42()  {
-        // read vcf v4.3 and try to write it to a vcf v4.2 (header is rejected)
-        final IOPath outputPath = IOUtils.createTempPath("rejectWrite43HeaderVCF", ".vcf");
-        readWriteVCFToPath(new HtsPath(VARIANTS_TEST_DIR + "variant/vcf43/all43Features.vcf"),
-                outputPath,
-                VCFCodecV4_3.VCF_V43_VERSION);
-    }
-
```

Comment:
Now that this case succeeds, the "vcfReadWriteTests" data provider at the top of this file should have a 4.3 file added to it (previously, we couldn't read in a 4.3 file and write it out as the "default" version, since the default was 4.2, but now that the default is 4.3, we can add that test case and remove the comment saying why its not there). The new test case would be:

`                { new HtsPath(VARIANTS_TEST_DIR + "variant/vcf43/all43FeaturesCompressed.vcf.gz"), VCFCodecV4_3.VCF_V43_VERSION },
`

Reply:
Additionally, there should be at least one new test that demonstrates that a 4.2 input that fails upgrade gets rejected by the VCF4.3 encoder.

There are also a bunch of other accidental things happening in this file now that we should clean up. Specifically, there are test inputs that are subjected to auto-upgrade now, but fail  (because they contain a PEDIGREE line that can't be upgraded - which nicely demonstrates the fallback case and the error message display, so thats good), but we'll need to revisit these tests in light of the changes on this branch to make sure they're still testing what they're supposed to be testing.

Perhaps I should add fix up this file and submit it for merging into your branch while you carry on with the other changes.